sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,217 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division
+
+import numpy as np
+from ..externals.six.moves import xrange
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
```

Comment:
What do you think about deferring this feature for future PRs? I feel like we can add a `strategy` key word argument to this constructor, which which might implement one or more of the above binning methods. (And perhaps percentile binning, if it's not included in this PR.)